### PR TITLE
Display print location information

### PR DESCRIPTION
**Pull request by aug24**
Print location information column is added to the workflow table.

Normal:
image
Compact:
image

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                                           externalData: Option[ExternalData]
+                                         ): Future[Option[String]] = {
+    externalData
+      .flatMap(e => Some(e.actualPublicationId, e.actualBookId, e.actualBookSectionId)) match {
+      case Some((Some(pId), Some(bId), Some(bsId))) => getPrintLocationDescriptionFromIds(contentAPI, pId, bId, bsId)
+      case _ => Future.successful(None)
+    }}
+
+  private def getPrintLocationDescriptionFromIds(contentAPI: ContentAPI, pId: Long, bId: Long, bsId: Long): Future[Option[String]] =
+    memoize(60.minutes) {
```
Comment (jonathonherbert) [inline common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala:63]: Discussed over chat but for posterity – memoizing at this level will produce a new cache key for every new set of arguments, even if some of those arguments have been seen within the cache window. For example, calls to the parent with pId, bId and bsId (1,2,3), (1,2,4) and (1,2,5) will all result in fresh invocations, although ids 1 and 2 are shared.

It's probably better to cache within `contentAPI.getTagInternalName`, which also has the benefit of hiding details of the cache implementation from consuming code.

Comment (aug24) [reply common-lib/src/main/scala/com/gu/workflow/api/StubAPI.scala]: Done.

**Comment by akash1810** [comment]
At first glance, this is doing more than reverting a commit - can we add more detail to the description please?

**Comment by aug24** [comment]
> At first glance, this is doing more than reverting a commit - can we add more detail to the description please?

Very good point. Done.

File: .java-version [hunk 2]
```diff
@@ -1 +1 @@
-1.8.0.242
+1.8.0.202
```
Comment (akash1810) [inline .java-version:1]: Why so low?

File: app/controllers/Api.scala [hunk 3]
```diff
@@ -319,3 +321,21 @@ object Api extends Controller with PanDomainAuthActions with SharedSecretAuth {
         block(req)
   }
 }
+
+//      def getTag(publicationId: Option[Int],
```
Comment (akash1810) [inline app/controllers/Api.scala:325]: Why are these commented out?

File: public/components/content-list-item/_content-list-item.scss [hunk 4]
```diff
@@ -541,3 +544,10 @@
         display: inline-block;
     }
 }
+
+.planned-print-location {
+    color: grey;
```
Comment (akash1810) [inline public/components/content-list-item/_content-list-item.scss:549]: Could we re-use something from the [palette](https://github.com/guardian/workflow-frontend/blob/master/public/layouts/global/_colours.scss)?

**Comment by akash1810** [comment]
Could we add some screenshots of the table in normal and compact mode?

File: app/controllers/Api.scala (hunk 3 above)
Comment (aug24) [reply app/controllers/Api.scala]: Carelessness?

**Comment by aug24** [comment]
> Not sure I understand/follow why we're moving from `CommonAPI` and `PrototypeAPI` to `StubAPI` 

Because _both_ those API class actually dealt with serving Stub information.

> and why we're caching?

For speed, mostly.

File: public/components/content-list-item/_content-list-item.scss (hunk 4 above)
Comment (aug24) [reply public/components/content-list-item/_content-list-item.scss:549]: Done.

File: .java-version (hunk 2 above)
Comment (aug24) [reply .java-version]: Changed to 232.

**Comment by jonathonherbert** [comment]
A bit more detail on the cacheing @akash1810 – because we need to get tags on each stub request (which happens often), because we expect content to have similar tags, and we don't expect that tags will change very often, cacheing tag information seemed like a good way of reducing the load on CAPI and improving the speed at which we serve requests.

It wouldn't be appropriate if we expected tag information to be extremely up to date.

**Comment by jonathonherbert** [comment]
Along those lines, @aug24 – what happens when CAPI is down? Do we have a reasonable timeout on the requests we make?

**Pull request merged**